### PR TITLE
fix: menu item classNames not work

### LINKED ISSUE #4119: [BUG] - DropdownItem ignore variants
### NextUI Version

~~2.4.2~~ 2.4.8

### Describe the bug

I created custom component <DropdownItem /> which extends variants ( extendVariants used ). When I use this component inside <DropdownSection /> everything is rendered correctly but when this component is a direct child of <DropdownMenu />   then <DropdownItem /> get variants but styles are ignored.

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Create custom Dropdown items for example
```
const DropdownItem = extendVariants(NextDropdownItem, {
  variants: {
    variant: {
      solid: {
        base: ['rounded-none'],
      },
    },
    size: {
      md: {
        base: 'py-4',
      }
    },
  },
  defaultVariants: {
    variant: 'solid',
    size: 'md',
  },
});
```
2. Put this component inside <DropdownMenu />

### Expected behavior

styles from custom component should be applied 

### Screenshots or Videos

_No response_

### Operating System Version

macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by winchesHe**
close #4119

## Summary by CodeRabbit

- **Bug Fixes**
	- Resolved issues with class names not functioning correctly for menu items in the menu component.

- **Tests**
	- Added new test cases to ensure proper functionality of class names for `MenuItem` components, verifying custom class name application and precedence handling.

- **Refactor**
	- Improved class name handling in the `Menu` component, allowing for more dynamic and flexible assignment of class names based on item properties.

### CODE REVIEW

File: .changeset/brave-trains-wave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/menu": patch
+---
+
+Fix menu item classNames not work
```
Comment (wingkwong) [inline .changeset/brave-trains-wave.md:5]: include issue number

File: packages/components/menu/src/menu.tsx [hunk 2]
```diff
@@ -47,11 +47,12 @@ function Menu<T extends object>(props: Props<T>, ref: ForwardedRef<HTMLUListElem
           hideSelectedIcon,
           ...item.props,
         };
+        const mergedItemClasses = itemProps.classNames ?? itemClasses;
```
Comment (wingkwong) [inline packages/components/menu/src/menu.tsx:50]: may double check if the expected behaviour is "merge" or "override". The naming suggest it's "merge" but the logic is "override". If it's the latter, we may also need to point that out in documentation.

**Review by wingkwong — COMMENTED** [review]
may double check if this PR is really related to the linked issue or not

File: packages/components/menu/src/menu.tsx (hunk 2 above)
Comment (winchesHe) [reply packages/components/menu/src/menu.tsx:50]: updated the documentation

**Review by wingkwong — COMMENTED** [review]
what about just call it `menuItemClasses` instead of `mergedItemClasses`? since either one will be used instead of merging them. The naming makes me feel `cn` is missing here while it's not.

File: .changeset/brave-trains-wave.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/menu": patch
```
Comment (jrgarciadev) [inline .changeset/brave-trains-wave.md:2]:
```suggestion
"@nextui-org/menu": patch
"@nextui-org/theme": patch
```

**Pull request merged**